src/utilities/format-market-cap.ts

```diff
@@ -0,0 +1,13 @@
+import { convertToCurrency } from './convert-to-currency';
+
+interface FormatMarketCap {
+  (marketCap: number): string | undefined; //FIXME: Why do I get an error when removing 'undefined' here
```

Comment:
because there are 4 `if`-s, but no `else`.
Basically your function could have a case where it doesn't return anything

But reading the code I can see that there are no real cases possible. Then I guess it's typescript being bad at these kinds of thinsg.

An easy fix would be to just do this at the end, without the `if`:
```typescript
return `${convertToCurrency(marketCap)}`;```

Reply:
Got it, thanks!